routemaster/state_machine/actions.py

```diff
@@ -93,7 +97,7 @@ def process_retries(
     app: App,
     state_machine: StateMachine,
     action: Action,
-    process_one_job: Callable,
+    should_terminate: Callable[[], bool],
```

Comment:
I wonder if it would be worth sharing the `IsExitingCheck` type (maybe renamed as `ShouldTerminate` for generality) and using that here?